src/editors/_shared/hooks/DataTypes/AbstractDataType.js

```diff
@@ -117,14 +117,23 @@ class AbstractDataType {
    * @returns {ReactComponent} Text input for editing common strings
    */
   stringEditComponent(props, placeholder, parsedValue) {
```

Comment:
It seems to me this function is trying to do too much with too little information. Maybe creating a arrayEditComponent and objectEditComponent would be more appropriate? Or parametrize the function to save the values with the expected type?